### DISCUSSION TITLE
Workflow file does not work even after updating actions/checkout@v3 and actions/setup-node@v3 to v4

### DISCUSSION THREAD

**Discussion by irregularhound** [bcomnes/deploy-to-neocities#183] [Q&A] [answered]
Out of nowhere, my Neocities workflow file, which up to this point was the one [here](https://scumsuck.com/resources/deployneocities/), stopped working. It has to do with something regarding Node upgrading from version 16 to version 20. I've gone to this GitHub action to replace the workflow file with [this one](https://github.com/bcomnes/deploy-to-neocities?tab=readme-ov-file#usage). After looking around and trying to figure out why it wasn't working, I was able to at least update the actions/checkout@v3 and actions/setup-node@v3 to the 4th version, and use the 2.0.2 version of the GitHub action.
The problem is I still can't deploy my website, and it's because of this error:
![Screenshot 2024-04-30 192315](https://github.com/bcomnes/deploy-to-neocities/assets/101528411/859e4ed8-b3c0-4901-a1de-77b0dfefd365)
I have no idea on what I'm doing here, and some advice on how to fix this would be appreciated.

**Comment by bcomnes** [comment]
The npm install step failed. Did you skip the checkout step? Does your project even need to install npm dependencies? It seems like you modified for your action file without considering what individual steps it was taking, and now it's doing things differently or things it doesn't need to be doing.

The deploy-to-neocities reusable GitHub action step updated the internal node runtime recently, but this will have almost no effect on user action code. 

Send me a URL to the failed action run and I can advise further. Screenshots provide almost no context.

**Reply by bcomnes** [reply]
I converted this to a discussion because this doesn't appear to be a bug directly.

**Reply by irregularhound** [reply]
Thank you. I am not well versed in this kind of thing, so forgive me for making any obvious mistakes. The only thing I did was replace the  actions/checkout@v3 and actions/setup-node@v3 to v4. Otherwise, I used the action file template shown in your Readme. 
I remember when I was trying to get it to work, I did remove the npm installation process, and that seemed to make it work slightly better, albiet still not deploying the website.
I made my repository public, so the action URL should be [here](https://github.com/AstroWildcat/bibliohound/actions/runs/8914264989/job/24481491522).

**Reply by bcomnes** [reply]
It says empty repo

**Reply by irregularhound** [reply]
I had to reupload the entire repository because my IRL information was attached to it. [Here's the new one.](https://github.com/AstroWildcat/bibliohound/actions/runs/8914773969/job/24485052160)

**Reply by bcomnes** [reply]
I sent in a PR https://github.com/AstroWildcat/bibliohound/pull/1

**Reply by bcomnes** [reply, marked as answer]
Ok now that you merged it, its trying to deploy, but the neocities API is rejecting some of the files. I need to improve the feedback on this, but since files get uploaded in a batched stream, its sort of hard to pin down. 

Best you can do now is to try and remove some files from your public folder and see if it works without the ones you removed. If it still doesn't work, remove move. Then once it works, add back files in chunks to try and figure out which ones neocities is rejecting.

**Reply by irregularhound** [reply]
Thank you. Sorry for the delay, but deleting my entire graphics folder seemed to do the trick.

**Reply by bcomnes** [reply]
One or more file in there is the problem. Try breaking it in half tonight narrow it down.
